crypto/fipsmodule/sha/internal.h

```diff
@@ -92,6 +103,26 @@ void sha512_block_data_order(uint64_t *state, const uint8_t *in,
 #define KECCAK1600_ASM
 #endif
 
+// SHA256_Init_from_state is a low-level function that initializes |sha| with a
+// custom state. |h| is the hash state in big endian. |n| is the number of bits
+// processed at this point. It must be a multiple of |SHA256_CBLOCK*8|.
+// It returns one on success and zero on programmer error.
+// External users should never use directly this function.
+OPENSSL_EXPORT int SHA256_Init_from_state(
+    SHA256_CTX *sha, const uint8_t h[SHA256_CHAINING_LENGTH], uint64_t n);
```

Comment:
Do we need to pass the value `n` or we can assume it's exactly one block that was processed? Will there be other use-cases? Which is more error prone if we call this function internally elsewhere; passing it or assuming it?

Reply:
For HMAC precompute, `n` will indeed always be 1. I made this function slightly more generic than needed for HMAC precompute in case we need it for other purposes. Also, I thought it would be a bit strange to have a function that only work for `n=1`. (I was not even sure how to name it in that case: `SHA256_Init_from_state_after_one_block`?)

Reply:
Is it ok for me to resolve this comment?